app/api/db/__init__.py

```diff
@@ -1,60 +1,159 @@
+`from api import db` in order to avoid confusion with instances of `DB`
+which are named `db` by convention.
+
+Example:
+    import api.db
+
+    db = api.db.init_db()
+
+    # non-ORM style usage
+    with db.get_connection() as conn:
+        conn.execute(...)
+
+    # ORM style usage
+    with db.get_session() as session:
+        session.query(...)
+        with session.begin():
+            session.add(...)
+
+Alternatively, if you are only importing the module for type hints (and
+to explicitly indicate dependencies), you can do `from api.db import DB`.
+
+Example:
+    from api.db import DB
+
+    @app.route("/health")
+    def health() -> Response:
+        db: DB = app.extensions["db"]
+        with db.get_connection() as conn:
+            conn.execute(...)
+"""
+
 import os
 import urllib.parse
-from contextlib import contextmanager
-from typing import Any, Generator, Optional
+from typing import Any, Optional
 
 import psycopg2
+import sqlalchemy
 import sqlalchemy.pool as pool
-from sqlalchemy import create_engine
-from sqlalchemy.engine import Engine
-from sqlalchemy.orm import scoped_session, sessionmaker
+from flask import Flask
+from sqlalchemy.orm import session
 
 import api.logging
 from api.db.db_config import DbConfig, get_db_config
-from api.db.migrations.run import have_all_migrations_run
+
+FLASK_EXTENSION_KEY = "db"
+
+# Re-export the Connection type that is returned by the get_connection() method
+# to be used for type hints.
+Connection = sqlalchemy.engine.Connection
+
+# Re-export the Session type that is returned by the get_session() method
+# to be used for type hints.
+Session = session.Session
 
 logger = api.logging.get_logger(__name__)
 
 
-def init(
-    config: Optional[DbConfig] = None,
-    check_migrations_current: bool = False,
-) -> scoped_session:
-    logger.info("connecting to postgres db")
-
-    engine = create_db_engine(config)
-    conn = engine.connect()
-
-    conn_info = conn.connection.dbapi_connection.info  # type: ignore
-    logger.info(
-        "connected to postgres db",
-        extra={
-            "dbname": conn_info.dbname,
-            "user": conn_info.user,
-            "host": conn_info.host,
-            "port": conn_info.port,
-            "options": conn_info.options,
-            "dsn_parameters": conn_info.dsn_parameters,
-            "protocol_version": conn_info.protocol_version,
-            "server_version": conn_info.server_version,
-        },
-    )
-    verify_ssl(conn_info)
-
-    # Explicitly commit sessions — usually with session_scope. Also disable expiry on commit,
-    # as we don't need to be strict on consistency within our routes. Once we've retrieved data
-    # from the database, we shouldn't make any extra requests to the db when grabbing existing
-    # attributes.
-    session_factory = scoped_session(
-        sessionmaker(autocommit=False, expire_on_commit=False, bind=engine)
-    )
+class DB:
```

Comment:
I like both of those names, do you prefer one over the other? if not i'll go with DBClient since it's slightly shorter and also consistent with other external services (and i think db can be thought of as a special case of another service, in this case RDS)